.github/workflows/branch-name-policy.yml

```diff
@@ -2,18 +2,18 @@ name: Fail when incorrect branch names are used
 on:
   push:
     branches:
-    - '*/*'
-    - '*/*/*'
+    - '*/**'
```

Comment:
So I did some pretty nasty research on this whole issue, and I'm _fairly_ confident that `*/**` will cover all cases. I'd be reasonably sure that the _minimatch_ package is being used here, and the `**/*` we had originally apparently matches anything anywhere, _including_ in the current directory:

`minimatch("foo", "**/*")` => `true`

Whereas `*/**` does what we'd expect, match anything with a depth of >=1:

`minimatch("foo", "*/**")` => `false`
`minimatch("1/2", "*/**")` => `true`
`minimatch("1/2/3", "*/**")` => `true`
`minimatch("1/2/3/4", "*/**")` => `true`

The _glob_ package behaves a bit differently in this regard, but I'm pretty sure that one is not being used internally here, most probably just _minimatch_. Did raise an issue with _glob_ though: https://github.com/isaacs/node-glob/issues/453